torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1059,6 +1058,41 @@ void ProcessGroupNCCL::eagerConnectSingleDevice(at::Device device) {
   getNCCLComm(key, device, OpType::ALLREDUCE);
 }
 
+bool ProcessGroupNCCL::useNonblocking() {
+#ifndef NCCL_HAS_COMM_NONBLOCKING
+  return false;
+#endif
+  // Already parsed, return the cached value
+  if (useNonblocking_.has_value()) {
+    return useNonblocking_.value();
+  }
+  // Get environment variable.
+  auto nbEnv = c10::utils::check_env("TORCH_NCCL_USE_COMM_NONBLOCKING");
+
+  // 1st priority: Respect the user's setting
+  if (options_->config.blocking != NCCL_CONFIG_UNDEF_INT) {
+    useNonblocking_ = options_->config.blocking == 0;
+    goto print_and_return;
```

Comment:
goto! thats something i haven't seen in a while...

Reply:
Almost never see goto in c++. Could use 'else if'? Dijkstra's publication of why [Go-to statement considered harmful](http://www.cs.utexas.edu/users/EWD/ewd02xx/EWD215.PDF) .